### PR TITLE
Pass props through from HOC

### LINKED ISSUE #14: Pass props
https://github.com/thchia/react-oidc/blob/49293dacade4be128438c0b9d388b2b246e986a8/src/makeAuth/index.tsx#L100

Would it be ok to change:
```
<AuthenticatorContext.Provider value={this.state.context}>
  {WrappedComponent}
</AuthenticatorContext.Provider>
```

To:

```
<AuthenticatorContext.Provider value={this.state.context}>
  <WrappedComponent {...this.props} />
</AuthenticatorContext.Provider>
```

### PR DESCRIPTION
**Pull request by chriskolenko**
WrappedComponent has changed to a React.ComponentType

Fixes #14 

I haven't done much Typescript. Hoping React.ComponentType is correct. 

Output of `yarn build`
```
yarn run v1.17.3
$ rm -rf ./lib && tsc
Done in 2.24s.
```

### CODE REVIEW

**Comment by chriskolenko** [comment]
@thchia I can confirm it's working. All the props are coming through. 

I changed the tests from : <Child /> to Child. Updated the readme. Would you like me to create a CHANGELOG.md? The breaking change makes me sad. This will be a minor version bump correct? 0.4.1-alpha?

**Comment by thchia** [comment]
Thanks for this! I will take a closer look at it later. Regarding the CHANGELOG, don’t worry about it as I will create that before publishing.

File: src/makeAuth/index.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ function makeAuthenticator({
   placeholderComponent,
   signinArgs
 }: IMakeAuthenticatorParams) {
-  return <Props extends {}>(WrappedComponent: React.ReactNode) => {
+  return <Props extends {}>(WrappedComponent: React.ComponentType) => {
```
Comment (thchia) [inline src/makeAuth/index.tsx]: I believe this should be:

```suggestion
  return <Props extends {}>(WrappedComponent: React.ComponentType<Props>) => {
```
This means that the component being wrapped expects the same props as those that will be passed to the component returned by this HoC.

**Review by thchia — CHANGES_REQUESTED** [review]
Can you add a test case for this change. Feel free to either change the implementation of `Child` or create another component that renders something based on its props. Then you will be able to assert that the props have been passed through using `getByText`.

**Comment by chriskolenko** [comment]
Done. First time writing a test in react. Thanks for forcing me to learn. Was very easy.

File: src/makeAuth/index.test.tsx [hunk 2]
```diff
@@ -92,4 +94,19 @@ describe('makeAuthenticator', () => {
     fireEvent.click(button)
     waitForElement(() => expect(queryByText('Logout')).toBeNull())
   })
+  
+  it('has props passed through', async () => {
+    const userManagerConfig = {
+      getUserFunction: failedGetUser
```
Comment (thchia) [inline src/makeAuth/index.test.tsx]: Let's change this to the `successfulGetUser`.

```suggestion
      getUserFunction: successfulGetUser
```

Currently your test asserts that `'bar'` is **not** present, which is correct if `getUserFunction` fails, but it is also correct if your implementation of passing props is broken, so this test does not strictly check that the passing of props works.

After this, you will need to update line 110. I've made a suggestion there.

File: src/makeAuth/index.test.tsx [hunk 3]
```diff
@@ -92,4 +94,19 @@ describe('makeAuthenticator', () => {
...
+    } as any
+
+    const WithAuth = makeAuthenticator({
+      userManager: makeUserManager(userManagerConfig, MockUserManager),
+      placeholderComponent: <Placeholder />
+    })(ChildProps)
+    const { queryByText } = render(<WithAuth foo="bar" />)
+
+    await expect(expiredGetUser()).resolves.toEqual({ expired: true })
+    expect(queryByText('bar')).toBeNull()
```
Comment (thchia) [inline src/makeAuth/index.test.tsx]:
```suggestion
    getByText('bar')
```

File: src/makeAuth/index.test.tsx [hunk 4]
```diff
@@ -92,4 +94,19 @@ describe('makeAuthenticator', () => {
...
+  it('has props passed through', async () => {
+    const userManagerConfig = {
+      getUserFunction: failedGetUser
+    } as any
+
+    const WithAuth = makeAuthenticator({
+      userManager: makeUserManager(userManagerConfig, MockUserManager),
+      placeholderComponent: <Placeholder />
+    })(ChildProps)
+    const { queryByText } = render(<WithAuth foo="bar" />)
```
Comment (thchia) [inline src/makeAuth/index.test.tsx]:
```suggestion
    const { getByText } = render(<WithAuth foo="bar" />)
```

**Review by thchia — CHANGES_REQUESTED** [review]
Suggested to change the test case you added to the "happy path", instead checking that the props **are** passed.

I probably won't get back to this till the weekend, but chances are this will be published under `v0.5.0`.

`v0.4.0` never made it to a latest release, and in fact the changes in that release turned out to be untenable. To follow semver strictly, I will release these changes as `0.5.0` (or maybe just finally go to v1).

**Comment by chriskolenko** [comment]
All good to merge this sometime today?

**Comment by thchia** [comment]
Yes! Sorry I didn’t get any notification you’d done the changes.. will merge this in the next few hours.

**Review by thchia — APPROVED** [review]

**Comment by thchia** [comment]
@chriskolenko [done](https://github.com/thchia/react-oidc/releases/tag/v1.0.0)

**Pull request merged**